numpy/core/src/common/cblasfuncs.c

```diff
@@ -387,6 +387,10 @@ cblas_matrixproduct(int typenum, PyArrayObject *ap1, PyArrayObject *ap2,
                                                  *((double *)PyArray_DATA(ap1));
             }
             else if (ap1shape != _matrix) {
+                for (size_t i = 0; i < PyArray_SIZE(out_buf); i += 1) {
+                    ((double *)PyArray_DATA(out_buf))[i] = -0.0;
+                }
```

Comment:
Two thoughts I had here:

1) I'm a bit surprised the testsuite is happy with this (locally, at least)--I would have thought it may be risky not to use one of the C-API iterator approaches because of strides and so on?
2) If we're going to commit to this, we may want to replace the original `memset` entirely and hoist this farther up in the control flow--I went a conservative to the specific code path in question, at least initially.

Reply:
`cblas_matrixproduct` should only be reached with contiguous arrays, I think, so that is fine.